Tests/SymbolKitTests/SymbolGraph/SemanticVersion/ErrorTests.swift

```diff
@@ -0,0 +1,376 @@
+        assertThrowingOversizedValueError(atPosition: .prerelease, inIdentifier: "8\(UInt.max)",             whenEvaluating: try .init(major: 2, minor: 4, patch: 6, prerelease: "8\(UInt.max)"))
+        assertThrowingOversizedValueError(atPosition: .prerelease, inIdentifier: "\(UInt.max)\(UInt.max)",   whenEvaluating: try .init(major: 3, minor: 5, patch: 7, prerelease: "\(UInt.max)\(UInt.max)"))
+        assertThrowingOversizedValueError(atPosition: .prerelease, inIdentifier: sum("\(UInt.max)", "1"),    whenEvaluating: try JSONDecoder().decode(SymbolGraph.SemanticVersion.self, from: Data(#"{"major":9,"minor":8,"patch":7,"prerelease":"\#(sum("\(UInt.max)", "1"))"}"#.utf8)))
+        assertThrowingOversizedValueError(atPosition: .prerelease, inIdentifier: "3\(UInt.max)",             whenEvaluating: try JSONDecoder().decode(SymbolGraph.SemanticVersion.self, from: Data(#"{"major":6,"minor":5,"patch":4,"prerelease":"3\#(UInt.max)"}"#.utf8)))
+        assertThrowingOversizedValueError(atPosition: .prerelease, inIdentifier: "\(UInt.max)\(UInt64.max)", whenEvaluating: try JSONDecoder().decode(SymbolGraph.SemanticVersion.self, from: Data(#"{"major":2,"minor":1,"patch":0,"prerelease":"\#(UInt.max)\#(UInt64.max)"}"#.utf8)))
+        
+        XCTAssertNoThrow(try SymbolGraph.SemanticVersion(major: 1, minor: 2, patch: 3, prerelease: "\(UInt.max)"))
+        XCTAssertNoThrow(try JSONDecoder().decode(SymbolGraph.SemanticVersion.self, from: Data(#"{"major":7,"minor":8,"patch":9,"prerelease":"\#(UInt.max)"}"#.utf8)))
+        
+        //    MARK: build metadata
+        
+        XCTAssertNoThrow(try SymbolGraph.SemanticVersion(major: 1, minor: 2, patch: 3, buildMetadata: "\(UInt.max)"))
+        XCTAssertNoThrow(try SymbolGraph.SemanticVersion(major: 4, minor: 5, patch: 6, buildMetadata: String(sum("\(UInt.max)", "1"))))
+        XCTAssertNoThrow(try SymbolGraph.SemanticVersion(major: 7, minor: 8, patch: 9, buildMetadata: "\(UInt.max)0"))
+        XCTAssertNoThrow(try SymbolGraph.SemanticVersion(major: 1, minor: 2, patch: 3, buildMetadata: "\(UInt.max)\(Int.max)"))
+        XCTAssertNoThrow(try JSONDecoder().decode(SymbolGraph.SemanticVersion.self, from: Data(#"{"major":7,"minor":8,"patch":9,"buildMetadata":"\#(UInt.max)"}"#.utf8)))
+        XCTAssertNoThrow(try JSONDecoder().decode(SymbolGraph.SemanticVersion.self, from: Data(#"{"major":0,"minor":1,"patch":2,"buildMetadata":"\#(sum("\(UInt.max)", "1"))"}"#.utf8)))
+        XCTAssertNoThrow(try JSONDecoder().decode(SymbolGraph.SemanticVersion.self, from: Data(#"{"major":3,"minor":4,"patch":5,"buildMetadata":"\#(UInt.max)6"}"#.utf8)))
+        XCTAssertNoThrow(try JSONDecoder().decode(SymbolGraph.SemanticVersion.self, from: Data(#"{"major":7,"minor":8,"patch":9,"buildMetadata":"\#(UInt.max)\#(UInt.max)"}"#.utf8)))
+    }
+        
+    //    TODO: Add tests for the precedence of error-throwing.
+    
+    func assertThrowingEmptyIdentifierError(
```

Comment:
True, but I think it's better if we test that the error messages are as expected as well. Because minor mistakes such as typos do happen, and a bit of more test can help catch these errors.

Reply:
I think maybe we're saying the same thing here. I'm saying that what really matters to test with the errors are their error messages.

These tests do verify the error messages, which is what's important. My minor feedback was just that there are 5 very similar test helpers to assert the error messages of the 5 different errors.

Reply:
> there are 5 very similar test helpers to assert the error messages of the 5 different errors

This is a very good point. I'll try to combine them.